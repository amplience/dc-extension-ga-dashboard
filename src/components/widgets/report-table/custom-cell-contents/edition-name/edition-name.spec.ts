/* eslint-disable @typescript-eslint/no-empty-function */

import { getByTestId, render } from '@testing-library/svelte';
import { Edition, Hub } from 'dc-management-sdk-js';
import { tick } from 'svelte';
import { get } from 'svelte/store';
import type { ManagementSdkService } from '../../../../../services/management-sdk/management-sdk.service';
import {
  hub,
  managementSdkService,
} from '../../../../../stores/dynamic-content';
import EditionName from './edition-name.svelte';

describe('EditionName', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('should initially render with id', async () => {
    managementSdkService.set(({
      getAppLinkForResource: () => new Promise(() => {}),
    } as unknown) as ManagementSdkService);

    const { container } = render(EditionName, {
      editionId: 'EDITION_ID',
    });
    expect(getByTestId(container, 'loading-1')).toMatchSnapshot();
  });

  it('should render the link', async () => {
    const mockHub = new Hub();
    hub.set(mockHub);
    managementSdkService.set(({
      getAppLinkForResource: jest.fn(() => Promise.resolve('LINK')),
      client: {
        editions: {
          get: jest.fn(() =>
            Promise.resolve(new Edition({ name: 'EDITION_NAME' }))
          ),
        },
      },
    } as unknown) as ManagementSdkService);

    const { container } = render(EditionName, {
      editionId: 'EDITION_ID',
    });
    expect(getByTestId(container, 'loading-1')).toMatchSnapshot();
    await tick();
    expect(getByTestId(container, 'loading-2')).toMatchSnapshot();
    await tick();
    expect(getByTestId(container, 'loaded')).toMatchSnapshot();
    expect(container.firstChild).toMatchSnapshot();

    expect(get(managementSdkService).getAppLinkForResource).toBeCalledWith(
      mockHub,
      Edition,
      'EDITION_ID'
    );

    expect(get(managementSdkService).client.editions.get).toBeCalledWith(
      'EDITION_ID'
    );
  });

  it('should render fallback to the id if no app link is found', async () => {
    managementSdkService.set(({
      getAppLinkForResource: jest.fn().mockRejectedValue(new Error('test')),
    } as unknown) as ManagementSdkService);

    const { container } = render(EditionName, {
      editionId: 'EDITION_ID',
    });
    await tick();
    expect(getByTestId(container, 'error')).toMatchSnapshot();
  });
});
