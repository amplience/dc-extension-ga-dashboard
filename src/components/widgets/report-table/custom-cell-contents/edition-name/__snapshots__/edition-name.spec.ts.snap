// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditionName should initially render with id 1`] = `
<span
  data-testid="loading-1"
>
  EDITION_ID
</span>
`;

exports[`EditionName should render fallback to the id if no app link is found 1`] = `
<span
  data-testid="error"
>
  EDITION_ID
</span>
`;

exports[`EditionName should render the link 1`] = `
<span
  data-testid="loading-1"
>
  EDITION_ID
</span>
`;

exports[`EditionName should render the link 2`] = `
<span
  data-testid="loading-2"
>
  EDITION_ID
</span>
`;

exports[`EditionName should render the link 3`] = `
<span
  data-testid="loaded"
>
  EDITION_NAME
</span>
`;

exports[`EditionName should render the link 4`] = `
<div>
  <a
    class="svelte-1c0jp0i"
    href="LINK"
    target="_self"
  >
    <span
      data-testid="loaded"
    >
      EDITION_NAME
    </span>
  </a>
  
  
</div>
`;
